src/traces/funnelarea/plot.js

```diff
@@ -0,0 +1,306 @@
+                        // tex and regular text together
+                        s.attr('data-notex', 1);
+                    });
+
+                    sliceText.text(pt.text)
+                        .attr({
+                            'class': 'slicetext',
+                            transform: '',
+                            'text-anchor': 'middle'
+                        })
+                        .call(Drawing.font, determineInsideTextFont(trace, pt, gd._fullLayout.font))
+                        .call(svgTextUtils.convertToTspans, gd);
+
+                    // position the text relative to the slice
+                    var textBB = Drawing.bBox(sliceText.node());
+                    var transform;
+
+                    var x0, x1;
+                    var y0 = Math.min(pt.BL[1], pt.BR[1]);
+                    var y1 = Math.max(pt.TL[1], pt.TR[1]);
+
+                    x0 = Math.max(pt.TL[0], pt.BL[0]);
+                    x1 = Math.min(pt.TR[0], pt.BR[0]);
+
+                    transform = getTransformToMoveInsideBar(x0, x1, y0, y1, textBB, {
+                        isHorizontal: true,
+                        constrained: true,
+                        angle: 0,
+                        anchor: 'middle'
+                    });
+
+                    sliceText.attr('transform',
+                        'translate(' + cx + ',' + cy + ')' + transform
+                    );
+                });
+            });
+
+            // add the title
+            var titleTextGroup = d3.select(this).selectAll('g.titletext')
+                .data(trace.title.text ? [0] : []);
+
+            titleTextGroup.enter().append('g')
+                .classed('titletext', true);
+            titleTextGroup.exit().remove();
+
+            titleTextGroup.each(function() {
+                var titleText = Lib.ensureSingle(d3.select(this), 'text', '', function(s) {
+                    // prohibit tex interpretation as above
+                    s.attr('data-notex', 1);
+                });
+
+                var txt = fullLayout.meta ?
+                    Lib.templateString(trace.title.text, {meta: fullLayout.meta}) :
+                    trace.title.text;
+
+                titleText.text(txt)
+                    .attr({
+                        'class': 'titletext',
+                        transform: '',
+                        'text-anchor': 'middle',
+                    })
+                .call(Drawing.font, trace.title.font)
+                .call(svgTextUtils.convertToTspans, gd);
+
+                var transform = positionTitleOutside(cd0, fullLayout._size);
+
+                titleText.attr('transform',
+                    'translate(' + transform.x + ',' + transform.y + ')' +
+                    (transform.scale < 1 ? ('scale(' + transform.scale + ')') : '') +
+                    'translate(' + transform.tx + ',' + transform.ty + ')');
+            });
+        });
+    });
+};
+
+function move(a) {
```

Comment:
This thing is only called once. I would :hocho:

Reply:
Cut in https://github.com/plotly/plotly.js/pull/3876/commits/7e6b9aee160110a8d17f3be746f6ccf44531501d.